File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashSet<Integer> hash = new HashSet<>();
        Arrays.sort(arr);
        int occur = 1;

        for(int i = 1; i < arr.length; i++) {
            if (arr[i] == arr[i - 1]){
                occur++;
                if (i == arr.length-1) {
                    if (hash.contains(occur))
                        return false;
                }
            }
            else{
                if (hash.contains(occur))
                    return false;
                hash.add(occur);
                if (i == arr.length-1) {
                    if (hash.contains(1))
                        return false;
                }
                occur = 1;
            }
        }
        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.swift
class Solution {
    func uniqueOccurrences(_ arr: [Int]) -> Bool {
        var dict = [Int : Int]()
        
        for i in arr {
            if dict[i] == nil { dict[i] = 1 }
            else { dict[i]! += 1 }
        }
        
        var arr = Array(dict.values)
        arr = arr.sorted(by: <)
        var i = 0
        
        while i < arr.count-1 {
            if arr[i] == arr[i+1] { return false }
            i += 1
        }
        
        return true
    }
}